gnomad_qc/v3/variant_qc/random_forest.py

```diff
@@ -0,0 +1,625 @@
+            & (ht[f"ac_qc_samples_{group}"] == 2),
+            "fail_hard_filters": (ht.QD < 2) | (ht.FS > 60) | (ht.MQ < 30),
+        },
+        singleton=ht.ac_release_samples_raw == 1,
+        ac_raw=ht.ac_qc_samples_raw,
+        ac=ht.ac_release_samples_adj,
+        ac_qc_unrelated_raw=ht.ac_qc_samples_unrelated_raw,
+    )
+
+    ht = ht.repartition(n_partitions, shuffle=False)
+    if checkpoint_path:
+        ht = ht.checkpoint(checkpoint_path, overwrite=True)
+
+    if impute_features_by_variant_type:
+        ht = median_impute_features(ht, {"variant_type": ht.variant_type})
+
+    summary = ht.group_by("omni", "mills", "transmitted_singleton",).aggregate(
+        n=hl.agg.count()
+    )
+    logger.info("Summary of truth data annotations:")
+    summary.show(20)
+
+    return ht
+
+
+def train_rf(ht, args):
+    features = FEATURES
+    test_intervals = args.test_intervals
+    if args.no_inbreeding_coeff:
+        features.remove("InbreedingCoeff")
+
+    if args.vqsr_training:
+        vqsr_ht = get_filters(f"vqsr_{args.vqsr_type}", split=True).ht()
+        ht = ht.annotate(
+            vqsr_POSITIVE_TRAIN_SITE=vqsr_ht[ht.key].info.POSITIVE_TRAIN_SITE,
+            vqsr_NEGATIVE_TRAIN_SITE=vqsr_ht[ht.key].info.NEGATIVE_TRAIN_SITE,
+        )
+        tp_expr = ht.vqsr_POSITIVE_TRAIN_SITE
+        fp_expr = ht.vqsr_NEGATIVE_TRAIN_SITE
+    else:
+        fp_expr = ht.fail_hard_filters
+        tp_expr = ht.omni | ht.mills
+        if not args.no_transmitted_singletons:
+            tp_expr = tp_expr | ht.transmitted_singleton
+
+    if test_intervals:
+        if isinstance(test_intervals, str):
+            test_intervals = [test_intervals]
+        test_intervals = [
+            hl.parse_locus_interval(x, reference_genome="GRCh38")
+            for x in test_intervals
+        ]
+
+    ht = ht.annotate(tp=tp_expr, fp=fp_expr)
+
+    rf_ht, rf_model = train_rf_model(
+        ht,
+        rf_features=features,
+        tp_expr=ht.tp,
+        fp_expr=ht.fp,
+        fp_to_tp=args.fp_to_tp,
+        num_trees=args.num_trees,
+        max_depth=args.max_depth,
+        test_expr=hl.literal(test_intervals).any(
+            lambda interval: interval.contains(ht.locus)
+        ),
+    )
+
+    logger.info("Joining original RF Table with training information")
+    ht = ht.join(rf_ht, how="left")
+
+    return ht, rf_model
+
+
+
+
+def get_rf_runs(rf_json_fp: str) -> Dict:
+    """
+    Loads RF run data from JSON file.
+
+    :param rf_json_fp: File path to rf json file.
+    :return: Dictionary containing the content of the JSON file, or an empty dictionary if the file wasn't found.
+    """
+    if file_exists(rf_json_fp):
+        with hl.hadoop_open(rf_json_fp) as f:
+            return json.load(f)
+    else:
+        logger.warning(
+            f"File {rf_json_fp} could not be found. Returning empty RF run hash dict."
+        )
+        return {}
+
+
+def get_run_data(
```

Comment:
It's in all of them, but at this point they are all actually different `input_args` so I might change it to take in an `input_args` dict rather than all of the parameters and then put the rest in `gnomad_methods`.